src/types.jl

```diff
@@ -321,6 +321,78 @@ function System(;
                     loggers, force_units, energy_units)
 end
 
+# A wrapper for replicas containing related configuration
+mutable struct ReplicaWrapper{C, V, T, L}
+    index::Int64
+    coords::C
+    velocities::V
+    temp::T
+end
+
+# A system to be simulated using replica exchange simulation
+mutable struct ReplicaSystem{D, G, T, A, AD, PI, SI, GI, RL, RT, RI, B, NF, L, F, E} <: AbstractSystem{D}
+    atoms::A
+    atoms_data::AD
+    pairwise_inters::PI
+    specific_inter_lists::SI
+    general_inters::GI
+    replica_list::RL
+    replica_temps::RT
```

Comment:
I was thinking of generalizing this field to contain any reaction coordinate (be it temperature, interactions for hamiltonian REMD, pH etc.) Does this sound like an improvement?